codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -657,6 +643,32 @@ private State readHeaders(ByteBuf buffer) {
         }
     }
 
+    /**
+     * Invoked when a message with both a "Transfer-Encoding: chunked" and a "Content-Length" header field is detected.
+     * The default behavior is to <i>remove</i> the Content-Length field, but this method could be overridden
+     * to change the behavior (to, e.g., throw an exception and produce an invalid message).
+     * <p>
+     * See: https://tools.ietf.org/html/rfc7230#section-3.3.3
+     * <pre>
+     *     If a message is received with both a Transfer-Encoding and a
+     *     Content-Length header field, the Transfer-Encoding overrides the
+     *     Content-Length.  Such a message might indicate an attempt to
+     *     perform request smuggling (Section 9.5) or response splitting
+     *     (Section 9.4) and ought to be handled as an error.  A sender MUST
+     *     remove the received Content-Length field prior to forwarding such
+     *     a message downstream.
```

Comment:
> The phrase "OUGHT TO" conveys an optimistic assertion of an implementation behavior that is clearly morally right, and thus does not require substantiation.
> \- https://tools.ietf.org/html/rfc6919#section-4

This seems to imply that this is a general expectation of all implementations. Thus, not doing this seems to indicate a diversion from the RFC spec?

Reply:
From @joshbressers above:
> This is correct behaviour according to RFC 2616
>
> > If a message is received with both a Transfer-Encoding header field and a Content-Length header field, the latter MUST be ignored.

That RFC is obsoleted by RFC7230 which states that the correct behaviour is to throw an exception when both 'Content-Type' and 'Transfer-Encoding' are present.

Reply:
RE: Ought To:

-  https://tools.ietf.org/html/rfc6919#section-4

Ah, rfc6919 is an april fools RFC (ignore me on that point then).

Reply:
While we're here, per [RFC 8174](https://tools.ietf.org/html/rfc8174) the normative key words only have their normative effect when rendered in uppercase. Thus, even if "ought to" were a relevant normative key word, the fact that it has been rendered lowercase means it's not intended to have normative effect.